src/service/service.ts

```diff
@@ -795,33 +795,30 @@ export class Discv5 extends (EventEmitter as { new (): Discv5EventEmitter }) {
           this.enr.encodeToValues(this.keypair.privateKey);
         }
         nodes.push(this.enr);
+        if (nodes.length >= 16) {
+          break topLoop;
+        }
       } else {
-        nodes.push(...this.kbuckets.valuesOfDistance(distance));
-      }
-    }
-    // limit response to 16 nodes
-    nodes = nodes.slice(0, 16);
-    if (nodes.length === 0) {
-      log("Sending empty NODES response to %o", nodeAddr);
-      try {
-        this.sessionService.sendResponse(nodeAddr, createNodesMessage(id, 1, nodes));
```

Comment:
After this PR we no longer send a NODES response if we've found no nodes, is that intended?

Reply:
when there is no nodes, the `chunkify` function returns `[[]]` (see the unit test), in that case we still send 1 response with total = 1